proto-public/pbmesh/v1alpha1/upstreams.proto

```diff
@@ -5,96 +5,72 @@ syntax = "proto3";
 
 package hashicorp.consul.mesh.v1alpha1;
 
-import "google/protobuf/duration.proto";
 import "pbcatalog/v1alpha1/selector.proto";
-import "pbmesh/v1alpha1/connection.proto";
-import "pbmesh/v1alpha1/routing.proto";
+import "pbmesh/v1alpha1/upstreams_configuration.proto";
 import "pbresource/resource.proto";
 
 message Upstreams {
   // Selection of workloads these upstreams should apply to.
   // These can be prefixes or specific workload names.
   hashicorp.consul.catalog.v1alpha1.WorkloadSelector workloads = 1;
 
+  // upstreams is the list of explicit upstreams to define for the selected workloads.
   repeated Upstream upstreams = 2;
-  repeated PreparedQueryUpstream pq_upstreams = 3;
 
-  UpstreamConfig upstream_config = 4;
+  // pq_upstreams is the list of prepared query upstreams. This field is not supported directly in v2
+  // and should only be used for migration reasons.
+  repeated PreparedQueryUpstream pq_upstreams = 3;
 }
 
 message Upstream {
-  hashicorp.consul.resource.ID destination_ref = 1;
+  // destination_ref is the reference to an upstream service. This has to be pbcatalog.Service type.
+  hashicorp.consul.resource.Reference destination_ref = 1;
+
+  // destination_port is the port name of the upstream service. This should be the name
+  // of the service's target port.
   string destination_port = 2;
+
+  // datacenter is the datacenter for where this upstream service lives.
   string datacenter = 3;
 
+  // listen_addr is the address where Envoy will listen for requests to this upstream.
+  // It can provided either as an ip:port or as a Unix domain socket.
   oneof listen_addr {
-    TCPAddress tcp = 4;
+    IPPortAddress ip_port = 4;
     UnixSocketAddress unix = 5;
   }
-
-  UpstreamConfig upstream_config = 7;
 }
 
-message TCPAddress {
+message IPPortAddress {
+  // ip is an IPv4 address.
```

Comment:
If this is just a string, could this be IPv6?

Reply:
Yep, this could be IPv6 as this string is just passed to Envoy's `SocketAddress` which already supports both stacks

Reply:
Done!